source/constructs/api/data_source/service.py

```diff
@@ -936,8 +943,17 @@ def add_account(account_id: str):
         raise BizException(MessageEnum.SOURCE_ASSUME_ROLE_FAILED.get_code(),
                            MessageEnum.SOURCE_ASSUME_ROLE_FAILED.get_msg())
 
+def add_third_account(account):
+    crud.add_third_account(account)
+
+def delete_account(account_provider: str, account_id: str, region: str):
+    if account_provider == Provider.AWS:
+        delete_aws_account(account_id)
+    else:
+        delete_third_account(account_provider, account_id, region)
 
-def delete_account(account_id: str):
+
+def delete_aws_account(account_id):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 41 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 98% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.

Reply:
Will be resolved in later version.